lib/sycamore/sycamore/materialize.py

```diff
@@ -203,13 +234,21 @@ def local_source(self) -> list[Document]:
         logger.info(f"Using {self._orig_path} as cached source of data")
         if not self._fshelper.file_exists(self._success_path()):
             logging.warning(f"materialize.success not found in {self._orig_path}. Returning partial data")
+        from sycamore.utils.sycamore_logger import LoggerFilter
+
+        limited_logger = logging.getLogger(__name__ + ".limited_local_source")
+        limited_logger.addFilter(LoggerFilter())
```

Comment:
Is there a reason you define `limited_logger` at the top of the file and then redefine it here? Doesn't seem like you use it anywhere other than this method.

Reply:
Oops, it was only supposed to be down below. You need to have a separate limited logger for each thing you want to be separately rate limited.